controllers/datadogagent/feature/logcollection/feature.go

```diff
@@ -0,0 +1,200 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package logcollection
+
+import (
+	"strconv"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/DataDog/datadog-operator/apis/datadoghq/v1alpha1"
+	"github.com/DataDog/datadog-operator/apis/datadoghq/v2alpha1"
+	apiutils "github.com/DataDog/datadog-operator/apis/utils"
+
+	apicommon "github.com/DataDog/datadog-operator/apis/datadoghq/common"
+	apicommonv1 "github.com/DataDog/datadog-operator/apis/datadoghq/common/v1"
+	"github.com/DataDog/datadog-operator/controllers/datadogagent/feature"
+	"github.com/DataDog/datadog-operator/controllers/datadogagent/object/volume"
+)
+
+func init() {
+	err := feature.Register(feature.LogCollectionIDType, buildLogCollectionFeature)
+	if err != nil {
+		panic(err)
+	}
+}
+
+func buildLogCollectionFeature(options *feature.Options) feature.Feature {
+	logCollectionFeat := &logCollectionFeature{}
+
+	return logCollectionFeat
+}
+
+type logCollectionFeature struct {
+	enable                     bool
+	containerCollectAll        bool
+	containerCollectUsingFiles bool
+	containerLogsPath          string
+	podLogsPath                string
+	containerSymlinksPath      string
+	tempStoragePath            string
+	openFilesLimit             int32
+
+	owner metav1.Object
+}
+
+// Configure is used to configure the feature from a v2alpha1.DatadogAgent instance.
+func (f *logCollectionFeature) Configure(dda *v2alpha1.DatadogAgent) feature.RequiredComponents {
+	f.owner = dda
+	if dda.Spec.Features.LogCollection != nil && apiutils.BoolValue(dda.Spec.Features.LogCollection.Enabled) {
```

Comment:
do have a code more readable, I think we could define `logCollection := dda.Spec.Features.LogCollection` inside the if.